### PR TITLE
Use uRandom device for Athena and put it behind a flag in simbaini

### PR DESCRIPTION
**Pull request by sfc-gh-vreddy**
The fix includes using a /dev/urandom instead of /dev/random based on the flag in SIMBAINI (ODBC). 

This can be set using transferagent method.

### CODE REVIEW

File: cpp/FileTransferAgent.hpp [hunk 1]
```diff
@@ -59,6 +59,16 @@ class FileTransferAgent : public IFileTransferAgent
...
+  /**
+   * Set useUrand to true to use /dev/urandom device
+   * Set it to false to use /dev/random device
+   * @param useUrand
+   */
+  virtual void setRandomDeviceAsUrand(bool useUrand)
+  {
+    m_useDevUrand = useUrand;
+  }
+
```
Comment (sfc-gh-vreddy) [inline cpp/FileTransferAgent.hpp:71]: This setRandomDeviceAsUrand() is invoked by filetransferagent and is used from ODBC and applications directly. 

The other setRandomDev is internal func in cpp/FileMetadataInitializer.hpp which sets the real random device to be used and the Encryptor uses the device value from this func.

File: cpp/FileTransferAgent.cpp [hunk 2]
```diff
@@ -139,6 +145,7 @@ void Snowflake::Client::FileTransferAgent::initFileMetadata(std::string *command
   m_FileMetadataInitializer.setAutoCompress(response.autoCompress);
   m_FileMetadataInitializer.setSourceCompression(response.sourceCompression);
   m_FileMetadataInitializer.setEncryptionMaterials(&response.encryptionMaterials);
+  m_FileMetadataInitializer.setRandomDev(m_useDevUrand);
```
Comment (sfc-gh-vreddy) [inline cpp/FileTransferAgent.cpp:148]: setRandomDev is internal func in cpp/FileMetadataInitializer.hpp which sets the real random device to be used and the Encryptor uses the device value from this func.

The other setRandomDeviceAsUrand() is invoked by filetransferagent and is used from ODBC and applications directly.

**Comment by sfc-gh-stakeda** [comment]
How does the customer enable / disable the new random function? What is the simba parameter?

**Comment by sfc-gh-vreddy** [comment]
> How does the customer enable / disable the new random function? What is the simba parameter?

@sfc-gh-stakeda Here is the ODBC PR https://github.com/snowflakedb/snowflake-odbc/pull/108/files

SIMBA setting is `UseURandomDevice`

**Pull request merged**